packages/libp2p/src/dcutr/dcutr.ts

```diff
@@ -342,7 +412,8 @@ export class DefaultDCUtRService implements Startable {
       try {
         const ma = multiaddr(addr)
 
-        if (!this.isPublicAndDialable(ma)) {
+        // TODO: find a way to test around this without hardcoding process.env.NODE_ENV checks.
+        if (process.env.NODE_ENV !== 'test' && !this.isPublicAndDialable(ma) ) {
```

Comment:
I found that DCUtR process wasn't ever being attempted because the addresses when testing are all private.

```
libp2p:dcutr:A B had no dialable multiaddrs +1ms
```

Reply:
Yep, this is going to be fiddly to test, we might need a config option to dial private addresses marked with 'this is just for testing' or similar.

Reply:
added a simple boolean toggle and marked `@internal`